daphne_worker/src/config.rs

```diff
@@ -264,16 +254,52 @@ impl<D> DaphneWorkerConfig<D> {
         DurableConnector::new(&self.ctx.as_ref().expect("no route context configured").env)
     }
 
-    pub(crate) fn get_hpke_receiver_config_for(
+    pub(crate) fn kv(&self, binding: &str) -> Result<KvStore> {
+        self.ctx
+            .as_ref()
+            .ok_or_else(|| Error::RustError("route context does not exist".to_string()))?
+            .kv(binding)
+    }
+
+    pub(crate) async fn get_hpke_receiver_config_for(
         &self,
         hpke_config_id: u8,
-    ) -> Option<&HpkeReceiverConfig> {
-        for hpke_receiver_config in self.hpke_receiver_config_list.iter() {
-            if hpke_config_id == hpke_receiver_config.config.id {
-                return Some(hpke_receiver_config);
+    ) -> Result<Option<HpkeReceiverConfig>> {
```

Comment:
Let's move the code for generating the receiver config (if it doesn't exist) here. Then the return type for this function can be `Result<HpkeReceiverConfig>`.

Reply:
It turned out that this does not work because `can_hpke_decrypt` will always return `Ok(true)` which is not correct when a client sends a report share encrypted using an expired HPKE public key.